mesonbuild/environment.py

```diff
@@ -1071,26 +1077,21 @@ def get_compiler_system_dirs(self):
         out = out.split('\n')[index].lstrip('libraries: =').split(':')
         return [os.path.normpath(p) for p in out]
 
+    def need_exe_wrapper(self, for_machine: MachineChoice = MachineChoice.HOST):
+        value = self.properties[for_machine].get('needs_exe_wrapper', None)
+        if value is not None:
+            return value
+        return not self.machines[for_machine].can_run()
+
     def get_exe_wrapper(self):
-        if not self.cross_info.need_exe_wrapper():
+        if not self.need_exe_wrapper():
             from .dependencies import EmptyExternalProgram
             return EmptyExternalProgram()
         return self.exe_wrapper
 
-
-class CrossBuildInfo:
+class ConfigFile(dict):
```

Comment:
no, no, no, no, no, no. Do *not* sublcass builtin types. There are so many pitfalls involved, everything may seem right to you, but it will lead to no end of bugs. Just don't do it. Please continue to use composition, keep the self.config variable.

Reply:
How about I subclass `SafeConfigParser`?

Reply:
Ah I realized that the old version also used `SafeConfigParser`. They both now have the same transformations.